### PR TITLE
Remove unnecessary integration tests.

### PR DESCRIPTION
**Pull request by thefazzer**
**Remove unnecessary integration tests.**

- Removing CanOverrideAllPorts, CanOverrideOnlyApiPort
- These are simple set/retrieve tests on command line args & don't require node spin up
- Already covered in NodeSettingsTest unit tests
- NB:  There is a bug in both of these so if they **are** necessary we need to reinstate https://github.com/stratisproject/StratisBitcoinFullNode/pull/3413

Preferred: @fassadlr @dangershony

### CODE REVIEW

**Comment by fassadlr** [comment]
@thefazzer can you please list the tests which cover these scenarios?

**Comment by fassadlr** [comment]
I had a quick look and there are no tests which cover for instance, the overriding of API/RPC ports. You will need to write a test that sets these values in a string[] and then pass it to `NodeSettings` constructor. Then that instance can be passed to ApiSettings, RpcSettings etc. You can then check that the value on RpcSettings (for example) is the value that you passed into node settings' constructor.

File: src/Stratis.Bitcoin.Tests/NodeConfiguration/NodeSettingsTest.cs [hunk 1]
```diff
@@ -148,5 +148,35 @@ public void NodeSettings_CanReadMultiValueFromNone()
...
+
+        /// <summary>
+        /// Verifies API port value can be passed in on startup.
+        /// </summary>
+        [Fact]
+        public void NodeSettings_CanOverrideOnlyApiPort()
+        {
+            const int apiport = 12345;
+            var nodeSettings = new NodeSettings(networksSelector: Networks.Networks.Bitcoin, args: new[] {  $"-apiport={apiport}" });
+            string result = nodeSettings.ConfigReader.GetOrDefault("apiport", string.Empty);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Tests/NodeConfiguration/NodeSettingsTest.cs]: OK so you are on the right track here, however you need to now pass `NodeSettings` into `ApiSettings` and then test the overridden value there.  The same for the test below 💪

**Comment by thefazzer** [comment]
@fassadlr Slightly more expansive change.

File: src/Stratis.Bitcoin.Tests/NodeConfiguration/NodeSettingsTest.cs [hunk 2]
```diff
@@ -148,5 +151,48 @@ public void NodeSettings_CanReadMultiValueFromNone()
...
+
+        /// <summary>
+        /// Verifies API port value can be passed in on startup.
+        /// </summary>
+        [Fact]
+        public void NodeSettings_CanOverrideOnlyApiPort()
+        {
+            const int apiport = 12345;
+            var network = Networks.Networks.Bitcoin;
+            var nodeSettings = new NodeSettings(networksSelector: network, args: new[] {  $"-apiport={apiport}" });
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Tests/NodeConfiguration/NodeSettingsTest.cs]: @thefazzer we wanna possibly move away from network selector at some point, so can you perhaps just pass in `new BitcoinRegTest()` here? The same with the test below.

**Pull request merged**